src/main/java/org/myrobotlab/service/ProgramAB.java

```diff
@@ -929,17 +955,17 @@ public SearchPublisher getSearch() {
 
   @Override
   public void setLocale(String code) {
-    this.locale = new Locale(code);
+    Runtime.getInstance().setLocale(code);
   }
 
   @Override
   public String getLanguage() {
-    return locale.getLanguage();
+    return Runtime.getInstance().getLanguage();
   }
 
   @Override
   public Locale getLocale() {
-    return locale;
+    return Runtime.getInstance().getLocale();

```

Comment:
should be locale of the currently loaded bot.

Reply:
removed .. promoted .. normalized - no longer an issue